lib/wasi/src/bin_factory/exec.rs

```diff
@@ -166,17 +166,17 @@ pub fn spawn_exec_module(
                     start
                         .call(&mut store, &[])
                         .map_err(WasiRuntimeError::from)
-                        .map(|_| 0)
+                        .map(|_| Errno::Success)
                 } else {
                     debug!("wasi[{}]::exec-failed: missing _start function", pid);
-                    Ok(Errno::Noexec as u32)
+                    Ok(Errno::Noexec)
```

Comment:
General comment on the `u32 => Errno` code changes:

I initially wanted to say this is nice for the code, BUT exit codes are generic, any number is a valid exit code.
By using `Errno` (which doesn't have an `other(x)` variant we restrict this and conversion would fail at some point. 

But just in general `Errno` != exit code .

Reply:
Exit codes are unfortunately not generic when it comes to LIBC and besides if you look at the code (in practice) Errno == exit code. Either way we can't have them diverge without breaking backwards compatibility with LIBC. This was causing serious issues in `dash` and `bash`

Reply:
Sure, the exit codes need to be correct for certain failure conditions to propagate properly.

But code is still three to return any exit code it likes. 
So won't this  break code that relies on specific / non-standard exit codes?

Reply:
The unknown exit codes will be converted into Errno::Unknown if they are used by `proc_join` otherwise they will remain u32's

Reply:
@theduke this is a good compromise I think - it gives the strong typing needed for `proc_join` to stay in sync but also allows for arbitary exit codes
https://github.com/wasmerio/wasmer/pull/3663/commits/a41bf19da05b2d859720db14b980055d382a13fc